### ISSUE TITLE
Bug: Laser Priority issues + Possible Addition

### ISSUE THREAD

**Issue by TehTurk** [Frooxius/NeosPublic#841]
Currently there's a few situations where the laser priority in world can have some issues. I was wonder if 
A) It becomes something we can configure in world, and have world settings for priority. This would be neat for certain map functionalities too. 
Or 
B) A list we ourselves can refer to to iron out anymore issues. 

Currently:
- Toolshelves are ignored when an inspector is behind it, 
- Color Dialogs and Component Panels will also have the same issue. 
- Gizmos are also an issue as they pass through to the Inspector

**Comment by Frooxius** [comment]
I don't plan on making this configurable, either per world or elsewhere. The priority is handled by a context sensitive code and complex filtering and checking, so it can't be represented by a simple list.

I can tweak the priorities though. Toolshelves definitely should take higher priority. Not sure what do you mean by component panels or color panels? Those are generally taking priority over inspectors.

Not sure what do you mean by gizmos passing through to the inspector either?

**Comment by TehTurk** [comment]
Dang that's a shame, can you elaborate more on the context sensitive code and the filter and checking? As for the priorties lemme get photos when I'm home.

**Comment by H3BO3** [comment]
yeah, some of the passthrough to inspectors is kind of annoying. What he means is the color picker panels, currently you have to grip them from the side/top bars to move them. Material orbs are also something that end up in front of an inspector often, and I have to weirdly aim my laser at an angle to grab them without grabbing the inspector instead.

Also, if you grab an item with your palm in the world and your laser is touching your dash, it will grab both. I'm not sure if communication between the two areas is possible now, but perhaps that could be looked into.

**Comment by TehTurk** [comment]
Here's some photos 
![Neos_67whyJaoci](https://user-images.githubusercontent.com/26357446/89289070-a5613780-d624-11ea-93f7-a9a054040871.jpg)
Color Picker
![Neos_4ZPduKwpWI](https://user-images.githubusercontent.com/26357446/89289092-ac884580-d624-11ea-9759-8a58215e4fb8.jpg)
Secondary Laser Select Not Tacking Priority.
Grabbable Laser Not Taking Priority(This would solve alot of the issues tbh, I still don't quite understand the change for Inspectors Taking Priority)
(More so what I meant by Gizmos sorry if unclear, I had another issue in my head
![Neos_WMcdT6hWvO](https://user-images.githubusercontent.com/26357446/89289192-cd509b00-d624-11ea-8449-5f372afe6dc4.jpg)
Tool Shelf not taking Priority. 

I am unaware of H3B0's comments on the dash/grabbing.
